frontend/src/components/pages/VolunteerDashboard/ShiftCard.tsx

```diff
@@ -0,0 +1,129 @@
+import { Box, Button, Stack, Text } from "@chakra-ui/react";
+import { format, parse } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+
+import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import * as Routes from "../../../constants/Routes";
+import {
+  CheckInWithShiftType,
+  ScheduleWithShiftType,
+  ShiftType,
+} from "../../../types/VolunteerTypes";
+import CardField from "../../common/CardField";
+import { getShiftColor } from "./types";
+
+interface CheckInOrScheduleProps {
+  donorId?: string;
+  isPickup?: boolean;
+  pickupLocation?: string;
+  volunteerTime?: string;
+  startTime?: string;
+  startDate?: string;
+  notes: string;
+  type: ShiftType;
+}
+const VolunteerShiftCard = ({
+  shift,
+}: {
+  shift: CheckInWithShiftType | ScheduleWithShiftType;
+}): JSX.Element => {
+  const {
+    donorId,
+    isPickup,
+    pickupLocation,
+    volunteerTime,
+    startTime,
+    startDate,
+    notes,
+    type,
+  } = shift as CheckInOrScheduleProps;
+  const [businessName, setBusinessName] = useState<string>("");
+
+  const history = useHistory();
+
+  const dateLocal = () => {
+    if (startDate) {
+      return format(new Date(startDate), "EEE MMM dd, yyyy");
+    }
+    return startTime && format(new Date(startTime), "EEE MMM dd, yyyy");
```

Comment:
this should be `volunteerTime` I think? always treating `volunteerTime` as source of truth for volunteers to refer to instead of the start Time of the scheduling. I know these will be the same always but still:

Reply:
to my understanding, `volunteerTime` is just a time string, and not a date object? So we won't be able to get the date from that.

Reply:
yeah I guess it's fine then since `dateLocal` just shows the day and that'll always be the same